Firestore/core/src/firebase/firestore/remote/serializer.h

```diff
@@ -124,6 +124,12 @@ class Serializer {
    */
   static pb_bytes_array_t* EncodeBytes(const std::vector<uint8_t>& bytes);
 
+  /**
+   * Returns the database ID, such as
+   * `projects/{project_id}/databases/{database_id}`.
+   */
+  pb_bytes_array_t* EncodeDatabaseId() const;
```

Comment:
In Firestore's protocol/naming scheme "IDs" are unqualified things and "names" are qualified (see [proto comments](https://github.com/googleapis/googleapis/blob/master/google/firestore/v1/firestore.proto#L568)). In this case, for a project ID of "p" and a database ID of "d", the database name is "projects/p/databases/d".

This should be `EncodeDatabaseName`.